Popover Fixes

neonbhai opened the pull request:
<!-- If necessary, assign reviewers that know the area or changes well. Feel free to tag any additional reviewers you see fit. -->

### Details
<!-- Explanation of the change or anything fishy that is going on -->
We now confirm bank account deletion using a center alert modal style popover (this is bottom on mobile).

### Fixed Issues

$ https://github.com/Expensify/App/issues/29729
PROPOSAL: https://github.com/Expensify/App/issues/29729#issuecomment-1765674068


### Tests
**Prerequisite:** This feature is under the `canUseWallets` beta. Return true [here](https://github.com/Expensify/App/blob/4dc53c780973fef0f31a3667eebafdea0d7dc389/src/libs/Permissions.ts#L5) to enable beta features.

**Prerequisite:** A bank account should be added to the user account.

**Steps to add a bank Account:**

Make sure you are using the staging server (Go to Preferences -> Use Staging Server)

1. Click on Profile Icon -> Workspaces -> Click on a workspace
2. Click on Bank Account.
3. Click on Connect Online With Plaid
4. Choose Wells Fargo. Use the following credentials:
        - Username: user_good
        - Password: pass_good
        - Mobile Code: credential_good
5.    Select Plaid Saving.
6.    Select both additional information checkboxes. Click Continue.
7.    Accept terms. Click on Connect Account Information
8.    Click on Allow
9.    You come back to connect Account Page. Click `Save & Continue`





**Test Steps:**

1. Click on profile picture -> Wallet
2. In the bank accounts card, we have a list of bank accounts. Click on a bank account.
3. Click Delete.
4. Verify that the confirmation popover shows in the middle for wide screens, 
5. Verify that the confirmation popover is bottom docked in mobile screens

- [x] Verify that no errors appear in the JS console

### Offline tests
<!---
Add any relevant steps that validate your changes work as expected in a variety of network states e.g. "offline", "spotty connection", "slow internet", etc. Manual test steps should be written so that your reviewer and QA testers can repeat and verify one or more expected outcomes. If you are unsure how the behavior should work ask for advice in the `#expensify-open-source` Slack channel.
--->
Same as Tests Step.

### QA Steps
<!---
Add a numbered list of manual tests that can be performed by our QA engineers on the staging environment to validate that your changes work on all platforms, and that there are no regressions present.
Add any additional QA steps if test steps are unique to a particular platform.
Manual test steps should be written so that the QA engineer can repeat and verify one or more expected outcomes in the staging environment.

For example:
1. Click on the text input to bring it into focus
6. Upload an image via copy paste
7. Verify a modal appears displaying a preview of that image
--->
Same as Tests Step.

### PR Author Checklist
<!--
This is a checklist for PR authors. Please make sure to complete all tasks and check them off once you do, or else your PR will not be merged!
-->

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
        - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



https://github.com/Expensify/App/assets/64629613/7440bcde-413c-4c64-8f0a-fab5f69d2c1d







</details>

<details>
<summary>Android: mWeb Chrome</summary>




https://github.com/Expensify/App/assets/64629613/af036dfc-8aaa-4a97-a690-72fa2f4ae9a9




</details>

<details>
<summary>iOS: Native</summary>



https://github.com/Expensify/App/assets/64629613/d961b06c-9728-45b9-9ad8-ad993b9a05e3









</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/64629613/1b835c17-2db5-42b3-8e90-85ed694ff071







</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



https://github.com/Expensify/App/assets/64629613/d3eb1c66-05eb-43a0-a89e-3a541413325a




</details>

<details>
<summary>MacOS: Desktop</summary>


https://github.com/Expensify/App/assets/64629613/290370a0-3b8a-442d-938b-df166ceb17f5




</details>

neonbhai committed:
Add popover with overlay to confirm Bank Account deletion

Santhosh-Sellavel commented:
@neonbhai Can you complete the author checklist & fix the lint issue?

neonbhai committed:
lint fix

Santhosh-Sellavel commented:
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native & iOS: Native</summary>

<img width="693" alt="Screenshot 2023-11-14 at 10 09 37 PM" src="https://github.com/Expensify/App/assets/85645967/08db0b29-61e1-49ad-9ab1-d55d4ae06f27">

</details>

<details>
<summary>Android: mWeb Chrome  & iOS: mWeb Safari</summary>

<img width="766" alt="Screenshot 2023-11-14 at 10 11 17 PM" src="https://github.com/Expensify/App/assets/85645967/337f0c9c-5826-4f07-97c0-dae793cda012">

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

<img width="996" alt="Screenshot 2023-11-14 at 10 08 34 PM" src="https://github.com/Expensify/App/assets/85645967/f74dac65-4bd7-442f-92d7-8e478ac437b3">

</details>

<details>
<summary>MacOS: Desktop</summary>

<img width="995" alt="Screenshot 2023-11-14 at 10 08 50 PM" src="https://github.com/Expensify/App/assets/85645967/61169450-fab6-4de3-8947-f863c0ccd84f">

</details>

Santhosh-Sellavel commented:
@neonbhai 

Bug: Delete action is not performed

https://github.com/Expensify/App/assets/85645967/23a1d26a-f298-414e-85f6-7b6a32c12024

Santhosh-Sellavel reviewed (changes requested):
Fix this https://github.com/Expensify/App/pull/30882#issuecomment-1810677781

neonbhai committed:
Move onModalHide to ConfirmModal prop

neonbhai committed:
Improve Confirm Modal Visibility logic

neonbhai committed:
Improve State Variable name

We change showConfirmDeleteContent to showConfirmDeleteModal to better reflect its purpose

neonbhai committed:
Merge branch 'Expensify:main' into popover-changes

Santhosh-Sellavel commented:
Any update?

Santhosh-Sellavel commented:
bump @neonbhai

neonbhai committed:
Move ConfirmModal inside Popover

neonbhai committed:
Revert "Move ConfirmModal inside Popover"

This reverts commit d7ead9f51fb4409c5a0c99c0721db27e45d2a588.

neonbhai committed:
Move Confirmation Modal component

This moves ConfirmModal as child of Popover while preserving its visibility logic

neonbhai committed:
Merge branch 'Expensify:main' into popover-changes

neonbhai committed:
Seperate closing logic for Popover and Confirm Modal

neonbhai committed:
Speed up Modal Visibility Time

We use a check to hide the delete menu View when the Confirm Modal is visible.

neonbhai committed:
Fix Button onPress action

Santhosh-Sellavel commented:
@neonbhai Will this be ready for review tomorrow?

Santhosh-Sellavel commented:
Thanks will review it tomorrow, meanwhile please test carefully and ensure there are no regressions. Resolve merge conflicts if any thanks!

Santhosh-Sellavel commented on src/pages/settings/Wallet/WalletPage/WalletPage.js:
```diff
@@ -511,34 +507,31 @@ function WalletPage({bankAccountList, cardList, fundList, isLoadingPaymentMethod
+                            shouldShowCancelButton
+                            danger
+                            onModalHide={() => {
+                                hideDefaultDeleteMenu();
```
Why do we need to call `hideDefaultDeleteMenu` here? It does not look DRY

We already have this block to hide the menu on the payment method reset

https://github.com/Expensify/App/blob/0a4db44b171b91f4522b049bd836004db50a1498/src/pages/settings/Wallet/WalletPage/WalletPage.js#L297-L306

neonbhai committed:
Remove redundant state change.

This removes the redundant call to hide DefaultDeleteMenu from Confirm Modal's onModalHide prop.

Santhosh-Sellavel commented:
Can you fix lint please?

neonbhai committed:
Merge branch 'Expensify:main' into popover-changes

Santhosh-Sellavel commented:
Can we merge this one @jasperhuangg ?

OSBotify commented:
:hand: This PR was not deployed to staging yet because QA is ongoing. It will be automatically deployed to staging after the next production release.

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/7210859713) to staging by https://github.com/jasperhuangg in version: 1.4.13-0 🚀

platform | result
---|---
🤖 android 🤖|success ✅
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|failure ❌
🕸 web 🕸|success ✅

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/7210859713) to staging by https://github.com/jasperhuangg in version: 1.4.13-0 🚀

platform | result
---|---
🤖 android 🤖|success ✅
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅
🕸 web 🕸|success ✅

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/7256299572) to production by https://github.com/jasperhuangg in version: 1.4.13-8 🚀

platform | result
---|---
🤖 android 🤖|success ✅
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅
🕸 web 🕸|success ✅
